source/retryable-reads/tests/README.rst

```diff
@@ -0,0 +1,269 @@
+
+    db.runCommand({
+        configureFailPoint: "failCommands",
+        mode: "off"
+    });
+
+Network Error Tests
+===================
+
+Network error tests are expressed in YAML and should be run against a standalone,
+shard cluster, or single-node replica set. Until `SERVER-34943`_ is resolved,
+these tests cannot be run against a multi-node replicaset.
+
+
+.. _SERVER-34943: https://jira.mongodb.org/browse/SERVER-34943
+
+Test Format
+-----------
+
+Test Format
+===========
+
+Each YAML file has the following keys:
+
+- ``database_name`` and ``collection_name``: Optional. The database and collection to use
+  for testing.
+
+- ``data``: The data that should exist in the collection under test before each
+  test run.
+
+- ``tests``: An array of tests that are to be run independently of each other.
+  Each test will have some or all of the following fields:
+
+  - ``description``: The name of the test.
+
+  - ``clientOptions``: Optional, parameters to pass to MongoClient().
+
+  - ``failPoint``: Optional, a server failpoint to enable expressed as the
+    configureFailPoint command to run on the admin database.
+
+  - ``sessionOptions``: Optional, parameters to pass to
+    MongoClient.startSession().
+
+  - ``operation``: A document describing an operation to be
+    executed. The document has the following fields:
+
+    - ``name``: The name of the operation on ``object``.
+
+    - ``object``: The name of the object to perform the operation on. Can be
+      "database", "collection", or "client."
+
+    - ``collectionOptions``: Optional, parameters to pass to the Collection()
+      used for this operation.
+
+    - ``command_name``: Present only when ``name`` is "runCommand". The name
+      of the command to run. Required for languages that are unable preserve
+      the order keys in the "command" argument when parsing JSON/YAML.
+
+    - ``arguments``: Optional, the names and values of arguments.
+
+    - ``result``: The return value from the operation, if any. This field may
+      be a scalar (e.g. in the case of a count), a single document, or an array
+      of documents in the case of a multi-document read. If the operation is
+      expected to return an error, the ``result`` is a single document that has
+      one or more of the following fields:
+
+      - ``error``: Optional. If ``true``, the test should expect an error or
+        exception. Implicitly true if ``result`` contains ``errorContains``,
+        ``errorCodeName``, ``errorLabelsContain``, and/or
+        ``errorLabelsOmit``. If ``false``, the test should expect no error or
+        exception, even if ``result`` contains ``errorContains``,
+        ``errorCodeName``, ``errorLabelsContain``, and/or ``errorLabelsOmit``.
+
+      - ``errorContains``: Optional. A substring of the expected error message.
+
+      - ``errorCodeName``: Optional. The expected "codeName" field in the server
+        error response.
```

Comment:
Is this intended to be checked after the driver translates the numeric error code? I may be behind on other spec tests, but I don't recall seeing this field before, and I don't recall servers returning a "codeName" field in an `{ok:0}` error response.

Reply:
I beleive this field was first introduced in the transactions tests: https://github.com/mongodb/specifications/tree/master/source/transactions/tests
Admittedly, we're not using any of these fields, so I could drop them from the test plan, leaving only "error" defined.

Reply:
If `error` is defined, would it be a boolean field? IIRC, that's what the CRUD spec uses and I'm contemplating using the same in my transaction helper spec tests.

Reply:
Yes, error is now a boolean field in the latest version! https://github.com/mongodb/specifications/pull/431/files#diff-e6269564f9bd464986ef56193a952c88R131